mmdet3d/ops/furthest_point_sample/points_sampler.py

```diff
@@ -0,0 +1,144 @@
+import torch
+from torch import nn as nn
+from typing import List
+
+from .furthest_point_sample import (furthest_point_sample,
+                                    furthest_point_sample_with_dist)
+from .utils import calc_square_dist
+
+
+def get_sampler_type(sampler_type):
+    """Get the type and mode of points sampler.
+
+    Args:
+        sampler_type (str): The type of points sampler.
+            The valid value are "D-FPS", "F-FPS", or "FS".
+
+    Returns:
+        class: Points sampler type.
+    """
+    if sampler_type == 'D-FPS':
+        sampler = DFPS_Sampler
+    elif sampler_type == 'F-FPS':
+        sampler = FFPS_Sampler
+    elif sampler_type == 'FS':
+        sampler = FS_Sampler
+    else:
+        raise ValueError('Only "sampler_type" of "D-FPS", "F-FPS", or "FS"'
+                         f' are supported, got {sampler_type}')
+
+    return sampler
+
+
+class Points_Sampler(nn.Module):
+    """Points sampling.
+
+    Args:
+        num_point (list[int]): Number of points.
+        fps_mod_list (list[str]: Type of FPS method, valid mod
+            ['F-FPS', 'D-FPS', 'FS'], Default: ['D-FPS'].
+            F-FPS: using feature distances for FPS.
+            D-FPS: using Euclidean distances of points for FPS.
+            FS: using F-FPS and D-FPS simultaneously.
+        fps_sample_range_list (list[int]): Range of points to apply FPS.
+            Default: [-1].
+    """
+
+    def __init__(self,
+                 num_point: List[int],
+                 fps_mod_list: List[str] = ['D-FPS'],
+                 fps_sample_range_list: List[int] = [-1]):
+        super(Points_Sampler, self).__init__()
```

Comment:
We may need to check the consistency between `len(num_points)` `len(fps_mod_list)`